src/Windowing/Silk.NET.Windowing.Common/Internals/ViewImplementationBase.cs

```diff
@@ -120,14 +118,22 @@ public void Reset()
         // Game loop controls
         public double FramesPerSecond
         {
-            get => _renderPeriod <= double.Epsilon ? 0 : 1 / _renderPeriod;
-            set => _renderPeriod = value <= double.Epsilon ? 0 : 1 / value;
+            get => _renderPeriod < 1 ? 0 : Frequency / _renderPeriod;
+            set => _renderPeriod = value <= double.Epsilon ? 0 : (long) (Frequency / value);
         }
 
         public double UpdatesPerSecond
         {
-            get => _updatePeriod <= double.Epsilon ? 0 : 1 / _updatePeriod;
-            set => _updatePeriod = value <= double.Epsilon ? 0 : 1 / value;
+            get
+            {
+                var updatePeriod = Interlocked.Read(ref _updatePeriod);
```

Comment:
This read shouldn't be interlocked, it makes no sense if all writes to it are interlocked